packages/wallets/src/features/cfd/modals/MT5AccountAdded/MT5AccountAdded.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { FC, useCallback, useMemo } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useActiveWalletAccount, useJurisdictionStatus, useMT5AccountsList, usePOA, usePOI } from '@deriv/api';
+import { ModalStepWrapper, ModalWrapper, WalletButton, WalletButtonGroup } from '../../../../components';
+import { useModal } from '../../../../components/ModalProvider';
+import useDevice from '../../../../hooks/useDevice';
+import { THooks, TMarketTypes, TPlatforms } from '../../../../types';
+import { companyNamesAndUrls, MarketTypeDetails, PlatformDetails } from '../../constants';
+import { CFDSuccess } from '../../screens/CFDSuccess';
+
+type TProps = {
+    account?: THooks.CreateMT5Account;
+    marketType: TMarketTypes.SortedMT5Accounts;
+    platform: TPlatforms.All;
+};
+
+const MT5AccountAdded: FC<TProps> = ({ account, marketType, platform }) => {
+    const { data: activeWallet } = useActiveWalletAccount();
+    const { isMobile } = useDevice();
+    const history = useHistory();
+    const { getVerificationStatus, isSuccess } = useJurisdictionStatus();
+    const { data: mt5Accounts } = useMT5AccountsList();
+    const { getModalState, hide } = useModal();
+    const { data: poiData } = usePOI();
+    const { data: poaData } = usePOA();
+    const addedAccount = mt5Accounts?.find(acc => acc.login === account?.login);
+    const verificationStatus = getVerificationStatus(addedAccount?.landing_company_short, addedAccount?.status);
+    const poiService = poiData?.is_pending ? poiData?.previous?.service : poiData?.current?.service;
+    const marketTypeTitle =
+        marketType === 'all' && Object.keys(PlatformDetails).includes(platform)
+            ? PlatformDetails[platform].title
+            : MarketTypeDetails[marketType].title;
+    const isPOIVerified = poiData?.status === 'verified';
+    const isPOAVerified = poaData?.status === 'verified';
+    const selectedJurisdiction = getModalState('selectedJurisdiction');
+    const landingCompanyName = `(${
+        companyNamesAndUrls?.[selectedJurisdiction as keyof typeof companyNamesAndUrls]?.shortcode
+    })`;
+
+    const isDemo = activeWallet?.is_virtual;
+
+    const renderAccountSuccessButton = useCallback(
```

Comment:
consideration for future PRs: whenever I see large functions rendering content inside a component, I'm always wondering, if that isnt't just a sign, that there should be a separate component for it?

Reply:
I wanted to do it, but the the problem is `renderButton` is accepting function that return component instead of component itself. IMO, its better to change that prop to accept ReactNode instead. I want to change it but it will affect many places